pcdsdevices/usb_encoder.py

```diff
@@ -0,0 +1,64 @@
+"""
+Class to handle USB encoders
+"""
+
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal
+
+from .interface import BaseInterface
+
+
+class UsDigitalUsbEncoder(BaseInterface, Device):
+    """
+    Class for the USB encoder by US Digital.
+    https://www.usdigital.com/products/accessories/interfaces/usb/usb4
+
+    Parameters
+    ----------
+    prefix: str
+        Base PV of the USB encoder
+
+    name: str
+        USB encoder name
+
+    linked_axis: device
+        Instance of the axis on which the encoder is mounted
+    """
+    zero = Cpt(EpicsSignal, ':ZEROCNT', kind='omitted')
+    pos = Cpt(EpicsSignal, ':POSITION', kind='hinted')
+    scale = Cpt(EpicsSignal, ':SCALE', kind='config')
+    offset = Cpt(EpicsSignal, ':OFFSET', kind='config')
+
+    tab_whitelist = ['pos', 'set_zero', 'set_zero_with_axis',
+                     'scale', 'offset']
+
+    def __init__(self, prefix, *, name, linked_axis=None, **kwargs):
+        super().__init__(prefix, name=name, **kwargs)
+        self.linked_axis = linked_axis
+
+    def set_zero(self):
+        """ Resets the encoder counts to 0 """
+        self.zero.put(1)
+
+    def set_zero_with_axis(self):
+        """
+        Resets the encoder counts to 0 and sets the linked_axis offset
+        so that its position is also 0.
+        """
+        self.zero.put(1)
+        if (
+            self.linked_axis is not None or
+            hasattr(self.linked_axis, 'set_current_position')
```

Comment:
I think the logic here is still subtly off. Should this be `and`, or else any non-`None` object will pass as `True`? Or should the `is not None` check go away entirely since it's redundant with the `hasattr` check? (`None` won't pass the `hasattr`). Or should the `hasattr` check go away so the user is quickly made aware that they've passed the wrong (some non-motor) device into the class?

Reply:
Indeed, the `hasattr` should be enough here. I really just wanna make sure that people get warned if the axis they pass does not have a `set_current_position` method (which I am not even sure every motor has?)
Will fix that in yet another commit!